### DISCUSSION TITLE
URL problem

### DISCUSSION THREAD

**Discussion by roberto-tenil** [devlikeapro/waha#173] [Q&A] [answered]
WAHA is a sensational tool. When I run it locally, on my computer, everything works fine.

I'm trying to deploy using a specific subdomain for WAHA, like: waha.myserver.com.

```
     environment:
       # Environment variables from https://waha.devlike.pro/docs/how-to/config/
       - WHATSAPP_HOOK_URL=https://httpbin.org/post
       - WHATSAPP_HOOK_EVENTS=message
       - WHATSAPP_DEFAULT_ENGINE=WEBJS
       - WHATSAPP_API_PORT=80
       - WHATSAPP_API_HOSTNAME=http://waha.myserver.com
       - WHATSAPP_RESTART_ALL_SESSIONS=True

```
After running the service in docker-compose the log shows me that the service is running as below
`WhatsApp HTTP API is running on: http://[::1]:80`

However, it doesn't work, when I try to access the api, I get the error: ERR_CONNECTION_REFUSED.

Shouldn't the service be running at `http://waha.myserver.com:80`?

Thanks in advance for your help.

**Comment by roberto-tenil** [comment, marked as answer]
Got it.
```
     ports:
       - "80:3000/tcp"
     environment:
       # Environment variables from https://waha.devlike.pro/docs/how-to/config/
       - WHATSAPP_HOOK_URL=https://httpbin.org/post
       - WHATSAPP_HOOK_EVENTS=message
       - WHATSAPP_DEFAULT_ENGINE=WEBJS
       - WHATSAPP_API_PORT=3000
```

**Comment by allburov** [comment]
Hi, it's a great question, thank you for the answer!

Btw, you should set it with no http, this way ` WHATSAPP_API_HOSTNAME=waha.myserver.com`. It's important because if you're working with media you'll still get "http://localhost:3000" instead
So the best configuration should work like this
```
     ports:
       - "80:80/tcp"
     environment:
       # Environment variables from https://waha.devlike.pro/docs/how-to/config/
       - WHATSAPP_HOOK_URL=https://httpbin.org/post
       - WHATSAPP_HOOK_EVENTS=message
       - WHATSAPP_DEFAULT_ENGINE=WEBJS
       - WHATSAPP_API_PORT=80
       - WHATSAPP_API_HOSTNAME=waha.myserver.com
```

**Comment by allburov** [comment]
Also one suggestion if you publish API to the internet - you should use https connection. I usually put API behind ngnix reverse proxy that handles ssl connection and uses pure http to communicate with WAHA. https://github.com/devlikeapro/whatsapp-http-api/issues/42#issuecomment-1376975224

It's import to set up [security for WAHA](https://waha.devlike.pro/docs/how-to/security/) for both Swagger (with `WHATSAPP_SWAGGER_USERNAME WHATSAPP_SWAGGER_PASSWORD`) and api key with `WHATSAPP_API_KEY`.
